jenkinsfiles/RHEL_Check.groovy

```diff
@@ -0,0 +1,371 @@
+
+                regexpFilterText: '$action',
+                regexpFilterExpression: '(opened|reopened|synchronize)'
+        )
+    }
+
+    stages {
+        stage('Check_User_in_Org') {
+            agent {
+                label "master"
+            }
+            steps {
+                script {
+                    try {
+                        retry(2) {
+                            fill_task_name_description()
+                            def check_user_return = sh(script: "python3 /localdisk2/oneDPL_CI/check_user_in_group.py -u  ${env.User}", returnStatus: true, label: "Check User in Group")
+                            echo "check_user_return value is $check_user_return"
+                            if (check_user_return == 0) {
+                                user_in_github_group = true
+                            }
+                            else {
+                                user_in_github_group = false
+                                currentBuild.result = 'UNSTABLE'
+                            }
+                        }
+                    }
+                    catch (e) {
+                        fail_stage = fail_stage + "    " + "Check_User_in_Org"
+                        user_in_github_group = false
+                        echo "Exception occurred when check User:${env.User} in group. Will skip build this time"
+                        sh script: "exit -1", label: "Set Failure"
+                    }
+                }
+            }
+        }
+
+        stage('RHEL_Check') {
+            when {
+                expression { user_in_github_group }
+            }
+            agent { label "Debug_RHEL" }
+            stages {
+                stage('Git-monorepo'){
+                    steps {
+                        script {
+                            try {
+                                retry(2) {
+                                    githubStatus.setPending(this, "Jenkins/RHEL_Check")
+                                    if (fileExists('./src')) {
+                                        sh script: 'rm -rf src', label: "Remove Src Folder"
+                                    }
+
+                                    sh script: 'cp -rf /export/users/oneDPL_CI/oneDPL-src/src ./', label: "Copy src Folder"
+                                    sh script: "cd ./src; git config --local --add remote.origin.fetch +refs/pull/${env.PR_number}/head:refs/remotes/origin/pr/${env.PR_number}", label: "Set Git Config"
+                                    sh script: "cd ./src; git pull origin; git checkout ${env.Commit_id}", label: "Checkout Commit"
+                                    if (fileExists('./oneAPI-samples')) {
+                                        sh script: 'rm -rf oneAPI-samples', label: "Remove oneAPI-samples Folder"
+
+                                    }
+
+                                    sh script: 'cp -rf /export/users/oneDPL_CI/oneAPI-samples ./', label: "Copy oneAPI-samples Folder"
```

Comment:
Have you considered direct cloning of https://github.com/oneapi-src/oneAPI-samples?

Smth like: `git clone --depth 1 https://github.com/oneapi-src/oneAPI-samples.git`

Reply:
This is the same reason with last one that copy from locally for oneAPI-samples will save at least 20 to 30 minutes for each task considering currently network environment. We will switch back to use git clone when the network has been fixed or improved. Before that, we intend to use current method to download source code.

Reply:
Thank you! Let's finalize the discussion in the comment above.